net/nimble/host/include/host/ble_store.h

```diff
@@ -31,6 +31,12 @@ extern "C" {
 #define BLE_STORE_OBJ_TYPE_PEER_SEC     2
 #define BLE_STORE_OBJ_TYPE_CCCD         3
 
+/** Failed to persist record; insufficient storage capacity. */
+#define BLE_STORE_EVENT_OVERFLOW        1
+
+/** About to execute a procedure that may fail due to overflow. */
+#define BLE_STORE_EVENT_OVERFLOW_NEXT   2
```

Comment:
I don't think BLE_STORE_EVENT_OVERFLOW_NEXT is particularly unclear.  It is so named because the next store operation will overflow.  The user may confuse this event type with the other (BLE_STORE_EVENT_OVERFLOW) because the names are similar, but I think this is going to be hard to prevent.  After all, the meanings of the events are similar!

If there is a specific name you like, I'm happy to use it.  Personally, I am OK with the current one, or several of the suggestions.